pkg/occlient/templates.go

```diff
@@ -18,9 +22,124 @@ type CommonImageMeta struct {
+		switch {
+		case cond != nil && cond.Reason == "NewReplicationControllerAvailable":
+			return true
+
+		case cond != nil && cond.Reason == "ProgressDeadlineExceeded":
+			return true
+
+		case cond != nil && cond.Reason == "RolloutCancelled":
+			return true
+
+		case cond != nil && cond.Reason == "DeploymentConfigPaused":
+			return true
+
+		case config.Status.UpdatedReplicas < config.Spec.Replicas:
+			return false
+
+		case config.Status.Replicas > config.Status.UpdatedReplicas:
+			return false
+
+		case config.Status.AvailableReplicas < config.Status.UpdatedReplicas:
+			return false
+		}
+	}
+	return false
+}
+
+func IsConfigApplied(componentConfig config.LocalConfigInfo, dc *appsv1.DeploymentConfig) (isConfApplied bool) {
+	/*
+		TODO: Add checks for updation of:
+		* ref
+		* ports
+		* path
+	*/
+
+	cfgAppName := componentConfig.GetApplication()
+	cfgCmpName := componentConfig.GetName()
+	cfgSrcType := string(componentConfig.GetSourceType())
+	cfgMinCpu := componentConfig.GetMinCPU()
+	cfgMaxCpu := componentConfig.GetMaxCPU()
+	cfgMinMemory := componentConfig.GetMinMemory()
+	cfgMaxMemory := componentConfig.GetMaxMemory()
+	cfgCmpType := componentConfig.GetType()
+	cfgProject := componentConfig.GetProject()
+
+	_, cmpBuilderImgVersionedName := getComponentBuilderDetails(dc)
+	cmpContainer, err := FindContainer(dc.Spec.Template.Spec.Containers, dc.Name)
+	if err != nil {
+		glog.V(4).Infof("Container with name %s not found in passed dc", dc.Name)
+		return false
+	}
+	isConfApplied = (
+	// TODO: Use the  label exported in pkg/application/labels/labels.go but defer it for now so as to defer the cyclic dependency to a new PR
+	dc.Labels["app.kubernetes.io/name"] == cfgAppName &&
+		cmpBuilderImgVersionedName == resolveComponentType(cfgCmpType) &&
+		dc.Labels[componentlabels.ComponentLabel] == cfgCmpName &&
+		dc.Namespace == cfgProject &&
+		// TODO: Use the  label exported in pkg/component/component.go but defer it for now so as to defer the cyclic dependency to a new PR
+		dc.Annotations["app.kubernetes.io/component-source-type"] == string(cfgSrcType))
+	if (cfgMinCpu != "") && (cfgMaxCpu != "") {
+		cpuExpected := util.FetchResourceQuantity(corev1.ResourceCPU, cfgMinCpu, cfgMaxCpu, "")
+		isConfApplied = isConfApplied && (cmpContainer.Resources.Limits.Cpu().Cmp(cpuExpected.MaxQty) == 0)
+		isConfApplied = isConfApplied && (cmpContainer.Resources.Requests.Cpu().Cmp(cpuExpected.MinQty) == 0)
+	}
+	if (cfgMinMemory != "") && (cfgMaxMemory != "") {
+		memoryExpected := util.FetchResourceQuantity(corev1.ResourceMemory, cfgMinMemory, cfgMaxMemory, "")
+		isConfApplied = isConfApplied && (cmpContainer.Resources.Limits.Memory().Cmp(memoryExpected.MaxQty) == 0)
+		isConfApplied = isConfApplied && (cmpContainer.Resources.Requests.Memory().Cmp(memoryExpected.MinQty) == 0)
+	}
+	return
+}
+
+func generateSupervisordDeploymentConfig(commonObjectMeta metav1.ObjectMeta, commonImageMeta CommonImageMeta,
```

Comment:
comments

Reply:
Not sure I follow this func completely(it wasn't added by me) but I have tried to add as much I could